packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -218,6 +218,10 @@ const GridRow = React.forwardRef<
 
   const publishClick = React.useCallback(
     (event: React.MouseEvent<HTMLDivElement>) => {
+      // See https://github.com/mui/mui-x/issues/8042
+      if (!apiRef.current.rootElementRef?.current?.contains(event.target as HTMLElement)) {
```

Comment:
`event.stopPropagation()` is an option too, but I'm not sure if it's any better

Reply:
If the goal is also to stop event bubbling, I think `event.stopPropagation` might be a better choice as compared to `return;` which causes event bubbling as depicted in this [codesandbox](https://codesandbox.io/s/bitter-sun-zqh5zg?file=/src/Bar.js). (Comment out `event.stopPropagation()` and uncomment `return` to see log printed in the parent component)

Reply:
When the nested grid is in the portal, we cannot rely on the DOM structure to get the parent cell that contains the nested grid.
I've opened https://github.com/mui/mui-x/pull/9741 to solve the original pain reported in the issue, I haven't noticed any issues with propagation with or without Portal